### PR TITLE
Split json subcategories

### PR DESCRIPTION
**Pull request by ohadschn**
Includes https://github.com/4tals/LinksForIsrael/pull/212 (so no need to merge that one if merging this one)
Closes https://github.com/4tals/LinksForIsrael/issues/186

Here's the script I used for the conversion:
```ps
$ErrorActionPreference = 'Stop'

$categoryFolders = Get-ChildItem -Directory -LiteralPath "G:\GitHub\LinksForIsrael2\_data\links"
$categoryJsons = $categoryFolders | ForEach-Object { "$($_)/links.json" }

foreach ($categoryJson in $categoryJsons) 
{
    $category = ConvertFrom-Json (Get-Content $categoryJson -Raw)
    $categoryFolder = [System.IO.Path]::GetDirectoryName($categoryJson)
    Write-Host "Processing category: $($category.name)"

    $categoryNames = @()
    foreach ($subCategory in $category.subCategories) { 
        $subCategoryName = if ($subCategory.displayName -eq "") {"Main"} else {$subCategory.name}
        Write-Host "subcategory: $($subCategory.name) | calculated name: $subCategoryName"
        $categoryNames += $subCategoryName

        $subCategoryFolder = "$categoryFolder/$subCategoryName"
        Write-Host "Creating folder: $subCategoryFolder"
        New-Item -Type Directory $subCategoryFolder -Force

        $subCategory | ConvertTo-Json -Depth 32 | Out-File "$subCategoryFolder/links.json"
    }

    Write-Host "new subcategories: $categoryNames"
    $category.subCategories = $categoryNames
    $category | ConvertTo-Json -Depth 32 | Out-File $categoryJson
} 
```

### CODE REVIEW

File: _data/categories.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[
+  "defense",
+  "InitiativesAggregation" ,
+  "volunteering" ,
+  "tech-initiatives" ,
+  "EquipmentNFood" ,
```
Comment (bergden-resonai) [inline _data/categories.json:6]: Maybe a good opportunity to unify the convention here (I see at least 3)

Comment (bergden-resonai) [reply _data/categories.json:6]: Unless it causes backwards compatibility issue for the existing links, in which case probably best to handle in a separate PR

**Review by 4tal — CHANGES_REQUESTED** [review]
Now search is not functional.

File: _data/categories.json (hunk 1 above)
Comment (ohadschn) [reply _data/categories.json:6]: I totally agree on the need (there are more issues, for example using "Hebrish" like "Needarim"
However I wanted to keep this PR as format change only, opened: https://github.com/4tals/LinksForIsrael/issues/227

**Comment by ohadschn** [comment]
> Now search is not functional.

Fixed

**Review by 4tal — APPROVED** [review]

**Pull request merged**